org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring/impl/EditorDocumentUndoChange.java

```diff
@@ -66,7 +66,7 @@ protected ITextEditor getEditor() {
 			if (editor != null && Objects.equal(editor.getSite().getId(), editorID))
 				return (ITextEditor) editor;
 			else
-				return (ITextEditor) page.openEditor(editorInput, editorID);
+				return (ITextEditor) page.openEditor(editorInput, editorID, true, IWorkbenchPage.MATCH_INPUT | IWorkbenchPage.MATCH_ID);
```

Comment:
am not sure if activate should be true

Reply:
I think `activate==true` is correct here since we want to provide some context for the user.

Reply:
> if you have mutile editors then searching needs to be done on input and id to get the text editor here and not the editor on the same/file input only

I see. Makes sense. It's a side effect of not properly using the right document type. Good catch.